lib/crypto/aes.toit

```diff
@@ -269,27 +276,34 @@ class AesGcm:
     before the verification tag has been verified with a call to $verify.
   */
   add data/ByteArray -> ByteArray:
+    size += data.size
     if buffer_:
-      bytes := gcm_add_ gcm_aes_ data buffer_
-      if bytes:
-        result := buffer_[0..bytes]
+      number_of_bytes := gcm_add_ gcm_aes_ data buffer_
+      if number_of_bytes:
+        result := buffer_[0..number_of_bytes]
         buffer_ = data
         return result
     // Output buffer was too small.  Make one that is certainly big enough.
-    result := ByteArray data.size + BLOCK_SIZE_
-    bytes /int := gcm_add_ gcm_aes_ data result
-    if buffer_ == null or data.size > buffer_.size: buffer_ = data
-    return result[..bytes]
+    result := ByteArray
+        round_up data.size BLOCK_SIZE_
+    number_of_bytes /int := gcm_add_ gcm_aes_ data result
+    if not buffer_ or data.size > buffer_.size: buffer_ = data
+    return result[..number_of_bytes]
 
   /**
   Finishes encrypting.
   Can be called after $start and $add.
-  Returns the 16 byte verification tag.
+  Returns a two-element list with the last encrypted bytes and the verification
```

Comment:
Would it make sense to just return the ByteArray, and stating that the last 16 bytes are the tag?
If they are always concatenated afterwards anyway, we might as well keep them together.